File: TransCoder_tokenized_test_set_functions/python/FIND_REPEATING_ELEMENT_SORTED_ARRAY_SIZE_N.py
def findRepeatingElement ( arr, low, high ) :
    if low > high :
        return - 1
    mid = ( low + high ) / 2
    if ( arr [ mid ] != mid + 1 ) :
        if ( mid > 0 and arr [ mid ] == arr [ mid - 1 ] ) :
            return mid
        return findRepeatingElement ( arr, low, mid - 1 )
    return findRepeatingElement ( arr, mid + 1, high )

File: results/java/FIND_REPEATING_ELEMENT_SORTED_ARRAY_SIZE_N.cpp
int f_filled ( int arr [ ] , int low , int high ) {
  if ( low > high ) return - 1 ;
  int mid = ( low + high ) / 2 ;
  if ( arr [ mid ] != mid + 1 ) {
    if ( mid > 0 && arr [ mid ] == arr [ mid - 1 ] ) return mid ;
    return f_filled ( arr , low , mid - 1 ) ;
  }
  return f_filled ( arr , mid + 1 , high ) ;
}
